bindings/python/CMakeLists.txt

```diff
@@ -70,57 +39,29 @@ ENDIF(BUILD_WITH_LIBPYTHON)
-  #LIST(FILTER ${PROJECT_NAME}_PYTHON_HEADERS EXCLUDE REGEX "^algorithm/parallel")
-  #LIST(FILTER ${PROJECT_NAME}_PYTHON_SOURCES EXCLUDE REGEX "^algorithm/parallel")
+  LIST(FILTER ${PROJECT_NAME}_PYTHON_HEADERS EXCLUDE REGEX "^${SOURCE_HEADERS_ROOT}/multibody/pool")
+  LIST(FILTER ${PROJECT_NAME}_PYTHON_SOURCES EXCLUDE REGEX "^multibody/pool")
+  LIST(FILTER ${PROJECT_NAME}_PYTHON_HEADERS EXCLUDE REGEX "^${SOURCE_HEADERS_ROOT}/algorithm/parallel")
+  LIST(FILTER ${PROJECT_NAME}_PYTHON_SOURCES EXCLUDE REGEX "^algorithm/parallel")
 ELSE(NOT BUILD_WITH_OPENMP_SUPPORT)
   IF(NOT BUILD_WITH_HPP_FCL_SUPPORT)
-    LIST_FILTER("${${PROJECT_NAME}_PYTHON_HEADERS}" "^multibody/pool/geometry.hpp" ${PROJECT_NAME}_PYTHON_HEADERS)
-    LIST_FILTER("${${PROJECT_NAME}_PYTHON_SOURCES}" "^algorithm/parallel/geometry.cpp" ${PROJECT_NAME}_PYTHON_SOURCES)
-    #LIST(FILTER ${PROJECT_NAME}_PYTHON_HEADERS EXCLUDE REGEX "^multibody/pool/geometry.hpp")
-    #LIST(FILTER ${PROJECT_NAME}_PYTHON_SOURCES EXCLUDE REGEX "^algorithm/parallel/geometry.cpp")
+  LIST(REMOVE_ITEM ${PROJECT_NAME}_PYTHON_HEADERS
+    ${SOURCE_HEADERS_ROOT}/multibody/pool/geometry.hpp
+    )
+  LIST(REMOVE_ITEM ${PROJECT_NAME}_PYTHON_SOURCES
+    algorithm/parallel/geometry.cpp
+    )
   ENDIF(NOT BUILD_WITH_HPP_FCL_SUPPORT)
 ENDIF(NOT BUILD_WITH_OPENMP_SUPPORT)
 
-LIST(APPEND HEADERS ${${PROJECT_NAME}_PYTHON_HEADERS})
-
-# Headers of the Python bindings
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python")
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/spatial")
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/multibody")
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/multibody/joint")
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/multibody/pool")
-IF(BUILD_WITH_HPP_FCL_PYTHON_BINDINGS)
-  MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/multibody/fcl")
-ENDIF(BUILD_WITH_HPP_FCL_PYTHON_BINDINGS)
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/parsers")
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/serialization")
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/algorithm")
-IF(BUILD_WITH_OPENMP_SUPPORT)
-  MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/algorithm/parallel")
-ENDIF(BUILD_WITH_OPENMP_SUPPORT)
-MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/bindings/python/utils")
-SYMLINK_AND_INSTALL_HEADERS("${${PROJECT_NAME}_PYTHON_HEADERS}" "bindings/python")
-
-# Headers of the Python parser
-IF(BUILD_WITH_LIBPYTHON)
-  MAKE_DIRECTORY("${${PROJECT_NAME}_BINARY_DIR}/include/pinocchio/parsers/python")
-  SYMLINK_AND_INSTALL_HEADERS("${${PROJECT_NAME}_PARSER_PYTHON_HEADERS}" "")
-ENDIF(BUILD_WITH_LIBPYTHON)
```

Comment:
This code will install `bindings/python/parsers/python.hpp` in both `pinocchio/parsers` and `pinocchio/bindings/python/parsers`.

I understand that this allow to easily include the Python model parser in C++. But since this code need the user to link against the binding library, I find it a little misleading.

Of course, if we don’t do the same, back compatibility could be broken.
@nim65s what do you think ?

Reply:
we can install the real one on the good location, and a shim in the wrong one.
The shim would raise a warning, and then include the good.

Reply:
Done